packaging/greenboot/microshift-running-check.sh

```diff
@@ -148,60 +149,95 @@ fi
 log_failure_cmd "pod-list" "${OCGET_CMD} pods -A -o wide"
 log_failure_cmd "pod-events" "${OCGET_CMD} events -A"
 
-# Wait for any pods to enter running state
-echo "Waiting ${WAIT_TIMEOUT_SECS}s for any pods to be running"
-if ! wait_for "${WAIT_TIMEOUT_SECS}" any_pods_running ; then
-    echo "Error: Timed out waiting for any MicroShift pod to be running"
-    exit 1
-fi
+function deployment_ready() {
+    local -r ns="$1"
+    local -r dep="$2"
 
-# Wait for MicroShift core pod images to be downloaded
-for i in "${!PODS_NS_LIST[@]}"; do
-    CHECK_PODS_NS=${PODS_NS_LIST[${i}]}
+    local ready
+    local reps
 
-    echo "Waiting ${WAIT_TIMEOUT_SECS}s for pod image(s) from the '${CHECK_PODS_NS}' namespace to be downloaded"
-    if ! wait_for "${WAIT_TIMEOUT_SECS}" namespace_images_downloaded ; then
-        echo "Error: Timed out waiting for pod image(s) from the '${CHECK_PODS_NS}' namespace to be downloaded"
-        exit 1
-    fi
-done
+    ready=$(${OCGET_CMD} -n "${ns}" "${dep}" -o jsonpath='{.status.readyReplicas}')
```

Comment:
I'm wondering if `availableReplicas` would be a better fit, but currently we set `minReadySeconds` only for `router-default` Pod, so they work mostly the same (perhaps something in scope of Pod Probes epic)
```
   availableReplicas	<integer>
     Total number of available pods (ready for at least minReadySeconds)
     targeted by this deployment.

   readyReplicas	<integer>
     readyReplicas is the number of pods targeted by this Deployment with a
     Ready Condition.
```

Reply:
Good call, I changed both to look at the available count.